crypto/src/merkle_tree/utils.rs

```diff
@@ -22,17 +22,18 @@ pub fn parent_index(node_index: usize) -> usize {
 
 // The list of values is completed repeating the last value to a power of two length
 pub fn complete_until_power_of_two<T: Clone>(values: &mut Vec<T>) -> Vec<T> {
-    if values.len() == 1 {
-        return values.clone(); // Return immediately if there is only one element.
-    }
     while !is_power_of_two(values.len()) {
         values.push(values[values.len() - 1].clone());
     }
     values.to_vec()
 }
 
 pub fn is_power_of_two(x: usize) -> bool {
-    (x != 0) && ((x & (x - 1)) == 0)
+    if x == 1 {
+        false
+    } else {
+        (x != 0) && ((x & (x - 1)) == 0)
+    }
```

Comment:
Strictly speaking, this is not true. `1` is `2^0`, so it is indeed a power of `2`. That said, here's a simpler alternative:
```suggestion
    (x > 1) && ((x & (x - 1)) == 0)
```

Reply:
Because technically we're lying, I expect a comment clarifying that:
1. We need this to keep the smallest tree at two leaves count.
2. We couldn't find a better name that wasn't overly verbose to what the function really does.
3. The function is private (please make it private as well) and is only used to ensure the tree has a power of 2 number of leaves.
4. A little white lie here saves us a lot of edge cases all around the code.

Otherwise, in a few months we will look at this, see it lies, conclude it's a bug, and proceed to "fix" it causing unexplainable bugs elsewhere.

Reply:
Thanks @Oppen will address this now.